src/service/AnnounceService.ts

```diff
@@ -358,18 +358,39 @@ export class AnnounceService {
                 } else {
                     const signerAccount = await resolveMainAccount();
                     if (transactions.length == 1) {
-                        await this.announceSimple(
-                            signerAccount,
-                            transactions[0],
-                            providedMaxFee,
-                            minFeeMultiplier,
-                            generationHash,
-                            currency,
-                            transactionService,
-                            listener,
-                            ready,
-                            nodeAccount.name,
-                        );
+                        if (transactions[0] instanceof MultisigAccountModificationTransaction) {
+                            const multisigModificationTx = transactions[0] as MultisigAccountModificationTransaction;
+                            await this.announceAggregateBonded(
+                                signerAccount,
+                                () => transactions.map((t) => t.toAggregate(mainAccount)),
+                                (multisigModificationTx.addressAdditions || []).length + (multisigModificationTx.minApprovalDelta || 0),
+                                deadline,
+                                networkType,
+                                defaultMaxFee,
+                                providedMaxFee,
+                                minFeeMultiplier,
+                                cosigners,
+                                generationHash,
+                                currency,
+                                transactionService,
+                                listener,
+                                ready,
+                                nodeAccount.name,
+                            );
+                        } else {
+                            await this.announceSimple(
+                                signerAccount,
+                                transactions[0],
+                                providedMaxFee,
+                                minFeeMultiplier,
+                                generationHash,
+                                currency,
+                                transactionService,
+                                listener,
+                                ready,
+                                nodeAccount.name,
+                            );
+                        }
                     } else {
                         await this.announceAggregateComplete(
```

Comment:
does the modifyMultisig tx needs to be announced as a bonded transaction? Using transactions.length might not always be safe. What if modifyMultisig is part of the transactions array? what about using array.filter?